user.go

```diff
@@ -87,6 +87,15 @@ type user struct {
 	networks        []*network
 	nextUpstreamID  uint64 // use with atomic
 	downstreamConns []*downstreamConn
+
+	pendingListsLock sync.Mutex
```

Comment:
Note to self: all messages (upstream and downstream) are processed in a per-user goroutine (I should write down the goroutine architecture). This means that the only reason why we need this lock is to clear the pending lists when an upstream connection is closed.